substrate/frame/staking/src/weights.rs

```diff
@@ -685,37 +703,77 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+	/// Proof: `Staking::MinCommission` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::MaxValidatorsCount` (r:1 w:0)
+	/// Proof: `Staking::MaxValidatorsCount` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::Nominators` (r:2000 w:1000)
+	/// Proof: `Staking::Nominators` (`max_values`: None, `max_size`: Some(558), added: 3033, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::MinNominatorBond` (r:1 w:0)
+	/// Proof: `Staking::MinNominatorBond` (`max_values`: Some(1), `max_size`: Some(16), added: 511, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::MaxNominatorsCount` (r:1 w:0)
+	/// Proof: `Staking::MaxNominatorsCount` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::CurrentEra` (r:1 w:0)
+	/// Proof: `Staking::CurrentEra` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::ValidatorCount` (r:0 w:1)
+	/// Proof: `Staking::ValidatorCount` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::CounterForValidators` (r:0 w:1)
+	/// Proof: `Staking::CounterForValidators` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::Payee` (r:0 w:2000)
+	/// Proof: `Staking::Payee` (`max_values`: None, `max_size`: Some(73), added: 2548, mode: `MaxEncodedLen`)
+	/// Storage: `Staking::CounterForNominators` (r:0 w:1)
+	/// Proof: `Staking::CounterForNominators` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `VoterList::CounterForListNodes` (r:0 w:1)
+	/// Proof: `VoterList::CounterForListNodes` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
 	/// The range of component `v` is `[500, 1000]`.
 	fn get_npos_targets(v: u32, ) -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `979 + v * (50 ±0)`
-		//  Estimated: `3510 + v * (2520 ±0)`
-		// Minimum execution time: 2_641_258_000 picoseconds.
-		Weight::from_parts(382_882_595, 3510)
-			// Standard Error: 11_991
-			.saturating_add(Weight::from_parts(4_695_820, 0).saturating_mul(v.into()))
-			.saturating_add(T::DbWeight::get().reads(2_u64))
-			.saturating_add(T::DbWeight::get().reads((1_u64).saturating_mul(v.into())))
-			.saturating_add(Weight::from_parts(0, 2520).saturating_mul(v.into()))
+		//  Measured:  `2195`
+		//  Estimated: `3774990 + v * (3774 ±0)`
+		// Minimum execution time: 184_772_982_000 picoseconds.
+		Weight::from_parts(109_226_888_420, 3774990)
+			// Standard Error: 417_730
+			.saturating_add(Weight::from_parts(155_577_516, 0).saturating_mul(v.into()))
+			.saturating_add(T::DbWeight::get().reads(9010_u64))
+			.saturating_add(T::DbWeight::get().reads((9_u64).saturating_mul(v.into())))
+			.saturating_add(T::DbWeight::get().writes(7008_u64))
```

Comment:
9000 READs and 7000 WRITEs per iteration is a log. Not sure if this is expected. @Ank4n @gpestana

Reply:
good find. this is indeed an error: https://github.com/paritytech/polkadot-sdk/pull/6025/files#r1839635371

I didn't review the weight carefully I will do another review.

Reply:
I normally use [this tool](https://github.com/ggwpez/substrate-weight-compare) when comparing weights (have to run it locally when using forks).  
It has a CLI and web version. Outputs for [proof-size.pdf](https://github.com/user-attachments/files/17729542/SWC-Web-proof.pdf) and [ref-time.pdf](https://github.com/user-attachments/files/17729544/SWC-Web-time.pdf)